packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
```

Comment:
This should be fairly simple (you can just go `if (options.secret === this.secret) ...`).

However, I'm curious what the correct behavior should be if it is the same? Should we error out? Or just continue as normal, just maybe without calling `options.secret.attach(this)`?

Reply:
Good question:
1. We can throw an error (Nah!)
2. Continue as normal (Probably!)
3. Have a different method for rotation for the master credentials (requires method deprecation, and apart from cleaner API I don't see any other value 😕)

What do you think?

PS: this auto-attach thingy seems to be inconsistent across the different features.

Reply:
I'd suggest to fix the attachment in a different PR (together with #13719). WDYT?

Reply:
@skinny85 any comment?

Reply:
I'd go with nr 2 from your list above.